### ISSUE TITLE
Data seems to be stale because RWS returns multiple Tijdstippen

### ISSUE THREAD

**Issue by Kaboutersuper** [physje/waterinfo#6]
Since a couple of days (Nov 5), it seems that RWS data is stale, i.e. it seems that no new data is present. When I do the request manually, I see multiple Tijdstippen and one of them is recent. Unfortunately an older Tijdstip comes before the newer Tijdstip; I guess it is assumed in the code that the first Tijdstip is the most recent one. I never had problems with this: not in home assistant but also not in my own python code (which definitely assumes he first Tijdstip is the most recent one).

Anyone having also stale data?

See partial dump below for Nieuwegeins water temperature.

{'WaarnemingenLijst': [{'Locatie': {'Locatie_MessageID': 11820, 'Coordinatenstelsel': '25831', 'X': 644981.501325792, 'Y': 5766893.93481021, 'Naam': 'Nieuwegein', 'Code': 'NWGN'}, 'MetingenLijst': [{'Tijdstip': '**2024-11-05T09:00:00.000+01:00**', 'Meetwaarde': {'Waarde_Numeriek': 12.9}, 'WaarnemingMetadata': {'StatuswaardeLijst': ['Ongecontroleerd'], 'BemonsteringshoogteLijst': ['-999999999'], 'ReferentievlakLijst': ['NVT'], 'OpdrachtgevendeInstantieLijst': ['LMW_WATERTEMP'], 'KwaliteitswaardecodeLijst': ['00']}}, {'Tijdstip': '**2024-11-07T14:50:00.000+01:00**', 'Meetwaarde': {'Waarde_Numeriek': 12.5}, 'WaarnemingMetadata': {'StatuswaardeLijst': ['Ongecontroleerd'], 'BemonsteringshoogteLijst': ['-1.98'], 'ReferentievlakLijst': ['NAP'], 'OpdrachtgevendeInstan

**Comment by imc67** [comment]
Same kind of issue here, found this add-on this morning and after install and a few hours this is the result.

Look at the last data date-time and strange graph.

![Schermafbeelding 2024-11-10 om 20 01 17](https://github.com/user-attachments/assets/fdc46c80-64f8-41f1-b082-04ce05e42ebd)

**Comment by physje** [comment]
@Kaboutersuper I can reproduce your issue. The integration indeed assumes the first value in the array to be the correct one.
The strange thing is -for as far as I know- it only happens with your location & measurement. If I try another combination I just get 1 value back.
@imc67 Are you looking for the calculated water level in the past and future in Westkapelle (WESTKPLE | WATHTBRKD) or the water level at this moment in Westkapelle (WKAP | WATHTE)?

**Comment by imc67** [comment]
@Kaboutersuper thanks for your very quick response! I’m looking indeed at least for the water level at this moment and expected  recorded archived records so I can see what levels it had and predict what it would be. However it would be perfect if the add-on could show waterlevel in past, present and future as these are available on the website of rws.

**Comment by Kaboutersuper** [comment]
@physje , thanks for the confirmation. I see the same issue for temperature at Brienenoordbrug NAP-6.5 m, Krimpen IJssel NAP -4.0m and Krimpen IJssel NAP -6.5m.

On the RWS watertemperatuur website  I see grey blocks for Nieuwegein and Brienenoordbrug NAP-6.5 m, but not for Krimpen IJssel. So the rws website seems to have trouble from the same problem, but not consistently.

Looking further on the website, I also guess that "Buitenbanken west" in the North sea has the same issue, given it is a grey block and the last measurement date on the RWS website (https://waterinfo.rws.nl/#/publiek/watertemperatuur?view=map) is also 05-11-2024 09:00.

Just guessing wildly, could it have anything to do with the new beta that they released around the same time that the problems started?

**Comment by Kaboutersuper** [comment]
@physje , I called RWS to notify them and they will call me back hopefully.

**Comment by petert1964** [comment]
@Kaboutersuper  Yep I see same issue.
![image](https://github.com/user-attachments/assets/ca13d24b-5402-4e54-9c7a-0d0f23e11676)

Very old data.

**Comment by stefbontinckx** [comment]
I'm receiving timeout errors as of a few days ago; 

raise ReadTimeoutError(
urllib3.exceptions.ReadTimeoutError: HTTPSConnectionPool(host='waterwebservices.rijkswaterstaat.nl', port=443): Read timed out. (read timeout=5)

Server side issues, I presume?

**Comment by physje** [comment]
I get the same results.
It does indeed look like server issues.
I haven't figured out yet whether the server is down or they made (unannounced) changes to the API. I can't find anything about the latter online.

**Comment by Kaboutersuper** [comment]
Since last week, I also got timeouts, but today I got "Http/1.1 Service Unavailable". On waterinfo.rws.nl I also don't see watertemperatures anymore, so I guess the server is down. They are working on a new backend, but I don't see any message that this has been deployed already (and even then our requests should not give a timeout).

Regarding my request to look into the double elements in latest-data: I got not response yet.

**Comment by Kaboutersuper** [comment]
Finally answer from RWS (kindof):
In de aanloop naar de nieuwe API wordt er op de achtergrond al een en ander rechtgezet in deze metadatavelden. Eerder werd de bemonsteringshoogte alleen in de naam opgenomen (KRY55 = -5.5m). In de nieuwe API worden de locatienamen gegeneraliseerd. Daarom wordt de bemonsteringshoogte opgenomen in het veld waar dit voor bedoeld is. 
Dat betekent effectief dat er een andere bemonsteringshoogte opgenomen is; tot 5-11 9:00 was dit -999999999 tov NVT, sindsdien is deze verhoogd tot -5.5 tov NAP.

In other words: there can now be different measuring heights in the output for a single location, rather than that the location is actually a combination of the location and measuring height (e.g. Krimpen IJssel NAP -4.0m and Krimpen IJssel NAP -6.5m).

Their proposed solution is to specify the measuring height as part of the request:
{
    "LocatieLijst": [{
        "X": "608662.635917219",
        "Y": "5752971.50132443",
        "Code": "KRY55"}],
    "AquoPlusWaarnemingMetadataLijst": [{
        "AquoMetadata": {
            "Grootheid":{
                "Code": "T"
            }
        }, 
        "WaarnemingMetadata": {
            "BemonsteringshoogteLijst":["-5.5"]
        }
    }]
}

But actually I think it would then be easier to
1) take the one with the latest date
or
2) filter out the 'BemonsteringshoogteLijst': ['-999999999'] from the current output unless it is the only measurement.

There will be a new API soon, so I hope it is possible to anticipate on what's coming (they put the change with the biggest impact already live apparently).

**Comment by physje** [comment]
I am dependent on the python package maintainer to anticipate on the new API.
As soon as they have issued a new release I can adjust my component accordingly.
Have already figured out RWS will changes a few things in the API I have to adjust the component for.
Will also solve the multiple times problem in that adjustment.

**Comment by physje** [comment]
@Kaboutersuper Made a new release with bugfix for the multiple times issue

**Comment by Kaboutersuper** [comment]
@physje I have uodated and I now get the right temperatures.

**Comment by petert1964** [comment]
Yes mine is also working. But it shows that i have downloaded 1.1.1 and of
course restarted everything. But it still shows me 1.1.0 if I go to
integrations?

Kind regards,

Mit freundlichen Grüßen,

Peter Blijleven
T.M.S Hunter Douglas Europe B.V.

Mobile +31653224506


Op do 5 dec 2024 om 20:00 schreef Kaboutersuper ***@***.***>:

**Comment by physje** [comment]
@petert1964 I think that's due to the fact I forgot to update `manifest.json` with the 1.1.1 version-number.
